### ISSUE TITLE
Search stemming issues on some word/phrases

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/pressfreedomtracker.us#1751]
Sophie reported that the name "Milstreed" does not come up in a search for an incident that contains it multiple times:

https://pressfreedomtracker.us/all-incidents/?search=milstreed

Lee discovered that "Milstred" does work:

https://pressfreedomtracker.us/all-incidents/?search=milstred

This suggests perhaps a stemming or other index configuration issue, with it being indexed incorrectly to begin with.

**Comment by chigby** [comment]
I'm pretty confident this is stemming related. From the `psql` shell:

```
select to_tsquery('english', 'Milstreed');
 to_tsquery 
────────────
 'milstre'
(1 row)
```

One option is to configure wagtail to index our data using the "simple" configuration (as opposed to the "english" one):

```
select to_tsquery('simple', 'Milstreed');
 to_tsquery  
─────────────
 'milstreed'
(1 row)
```

However, this would remove _all_ stemming from our text search index, meaning, for example, a search for "education" would not match a document with the word "educate" in it. Stemming is one reason why using text search is nicer than using something simpler like substring matching: in theory it can know how English (or other languages) works and find information that does not exactly match your query. Disabling stemming completely is something that would enhance searches for problematic problematic proper nouns like "Milstreed" or "Paris" but de-hance searches that don't include those kinds of terms.

It seems like it's possible, though, to customize the text search process so that "normal" words are stemmed but special words are not. Queries are passed through multiple layers of [Postgresql dictionaries](https://www.postgresql.org/docs/13/textsearch-dictionaries.html) to find a match, and stemming can happen at any layer. 

[Synonym dictionaries](https://www.postgresql.org/docs/13/textsearch-dictionaries.html#TEXTSEARCH-SYNONYM-DICTIONARY) allow words to be added to, basically, a curated list of ones we want to remove stemming from. This might be cumbersome to continue to add terms to, and also might be difficult to determine a complete list of words we'd want to add, but using this would get us what we want.

Ispell dictionaries are another type. It may be possible to add a comprehensive ispell dictionary (from, e.g., the hunspell package, used by OpenOffice), and _remove_ the [snowball dictionary](https://www.postgresql.org/docs/current/textsearch-dictionaries.html#TEXTSEARCH-SNOWBALL-DICTIONARY) that serves as postgresql's "catch all" dictionary that stems the words algorithmically (which is what's responsible for mis-stemming our "Milstreed" term from earlier). This would require some investigation. Adding the hunspell dictionary is mostly straightforward (I think!), but removing the snowball dictionary may lead to some words being incorrectly removed from search queries (since there's no longer something to catch all that was not caught earlier in the dictionary pipeline).

**Comment by harrislapiroff** [comment]
I think this issue is basically resolved in #1755. It's possible there's other issues with stemming, but the one that prompted us to open this ticket is resolved right? In that case I think we can close this until we discover other issues. Do you agree @chigby?

**Comment by chigby** [comment]
I do believe this solved the current issue.
